lib/SIL/Utils/OSSALifetimeCompletion.cpp

```diff
@@ -389,6 +392,14 @@ void OSSALifetimeCompletion::visitAvailabilityBoundary(
   visitor.propagateAvailablity(result);
 
   visitor.visitAvailabilityBoundary(result, visit);
+
+  visitUsersOutsideLinearLivenessBoundary(
```

Comment:
It looks like this makes `AvailabilityBoundaryVisitor` a four step process. Its class-level comment is no longer replete.

Reply:
Added the new step to the class-level comment.  It does feel now like the factoring/naming here is not quite ideal (there's the availability boundary and then there are these infinite loop instructions), but we can tweak that later on.

Reply:
Clarified the top-level comment to indicate that `AvailabilityBoundaryVisitor` only does half of `OSSALifetimeCompletion::visitAvailabilityBoundary`.